### PR TITLE
feat(content): #225 Bookmarks filter on Timeline

### LINKED ISSUE #171: Refactor Header component
Right now the `Header` component is inside the `Base` component because it used to exist in both the `NewTabPage` and `TimelinePage`, however, it is now only required in the `TimelinePage`.

In https://github.com/mozilla/activity-streams/pull/170 we just added some code to hide it, but we should refactor to move the `Header` component inside `TimelinePage` instead.

### LINKED ISSUE #225: Bookmarks filter/page
We don't actually have a filter for bookmarks that works on the Timeline view; it needs to be implemented. It should use the `RECENT_BOOKMARKS` data

### PR DESCRIPTION
**Pull request by k88hudson**
This adds sub-routing to the Timeline view, and a bookmark filter for the Timeline layout.

For now this is basically the same layout as "All", but it may change in the future.

I also added the `Header` component to the `TimelinePage` page instead of `Base`.

Fixes #225 
Fixes #171

### CODE REVIEW

File: lib/ActivityStreams.js [hunk 1]
```diff
@@ -223,6 +223,7 @@ ActivityStreams.prototype = {
         baseUrl,
         `${baseUrl}#/`,
         `${baseUrl}#/timeline`,
+        `${baseUrl}#/timeline/bookmarks`
```
Comment (k88hudson) [inline lib/ActivityStreams.js]: This doesn't actually seem to be working but i'm assuming rlr is fixing this in https://github.com/mozilla/activity-streams/issues/223

File: content-src/components/TimelinePage/TimelineBookmarks.js [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+const {connect} = require("react-redux");
+const {GroupedActivityFeed} = require("components/ActivityFeed/ActivityFeed");
+const Spotlight = require("components/Spotlight/Spotlight");
+
+const TimelineBookmarks = React.createClass({
+  render() {
+    const props = this.props;
+    return (<div className="wrapper">
+      <Spotlight sites={props.Bookmarks.rows} />
+      <GroupedActivityFeed title="Just now" sites={props.Bookmarks.rows} length="20" />
```
Comment (pdehaan) [inline content-src/components/TimelinePage/TimelineBookmarks.js]: Should this be `length={20}`? (see #237)

File: content-src/components/TimelinePage/TimelineHistory.js [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+const {connect} = require("react-redux");
+const {GroupedActivityFeed} = require("components/ActivityFeed/ActivityFeed");
+const Spotlight = require("components/Spotlight/Spotlight");
+
+const TimelineHistory = React.createClass({
+  render() {
+    const {History} = this.props;
+    return (<div className="wrapper">
+      <Spotlight sites={History.rows} />
+      <GroupedActivityFeed title="Just now" sites={History.rows} length="20" />
```
Comment (pdehaan) [inline content-src/components/TimelinePage/TimelineHistory.js]: Should this be `length={20}`?

File: content-src/components/TimelinePage/TimelineBookmarks.js (hunk 2 above)
Comment (k88hudson) [inline content-src/components/TimelinePage/TimelineBookmarks.js]: :+1:

**Comment by pdehaan** [comment]
@k88hudson Does this also fix #171?

**Comment by k88hudson** [comment]
@pdehaan ah yes, it does!

**Comment by pdehaan** [comment]
LGTM locally, and the code looks good &mdash; but I'll let @oyiptong and @rlr take a look before pressing the Big Green Button(tm).

File: lib/PlacesProvider.js [hunk 4]
```diff
@@ -349,7 +350,7 @@ Links.prototype = {
                     LIMIT :limit`;
 
     let links = yield this.executePlacesQuery(sqlQuery, {
-                  columns: ["bookmarkId", "bookmarkTitle", "bookmarkGuid", "url", "title", "lastVisitDate", "frecency", "type", "lastModified", "favicon", "mimeType"],
+                  columns: ["bookmarkId", "bookmarkTitle", "bookmarkGuid", "bookmarkDateCreated", "url", "title", "lastVisitDate", "frecency", "type", "lastModified", "favicon", "mimeType"],
```
Comment (oyiptong) [inline lib/PlacesProvider.js]: let's add a test in test-PlacesProvider, just so we have some regression safety net

**Comment by k88hudson** [comment]
@oyiptong ok, tests updated

**Comment by oyiptong** [comment]
i'll defer to @rlr for the react component code, r+ on the query side of things

**Comment by rlr** [comment]
looking!

File: lib/ActivityStreams.js (hunk 1 above)
Comment (rlr) [inline lib/ActivityStreams.js]: :+1: I'll make sure that one works too.

**Comment by rlr** [comment]
Nice cleanup with header. New timeline component LGTM. r=me

**Pull request merged**